projects/distributed-tracing/src/shared/dashboard/widgets/trace-detail/data/api-trace-detail-data-source.model.ts

```diff
@@ -1,74 +1,24 @@
-import { Dictionary } from '@hypertrace/common';
-import { ARRAY_PROPERTY, Model, ModelProperty, PLAIN_OBJECT_PROPERTY } from '@hypertrace/hyperdash';
-import { EMPTY, Observable, of } from 'rxjs';
-import { mergeMap } from 'rxjs/operators';
-import { Trace, traceIdKey, traceTypeKey } from '../../../../graphql/model/schema/trace';
-import { SpecificationBuilder } from '../../../../graphql/request/builders/specification/specification-builder';
-import {
-  TraceGraphQlQueryHandlerService,
-  TRACE_GQL_REQUEST
-} from '../../../../graphql/request/handlers/traces/trace-graphql-query-handler.service';
-import { GraphQlDataSourceModel } from '../../../data/graphql/graphql-data-source.model';
+import { Model } from '@hypertrace/hyperdash';
+import { Trace } from '../../../../graphql/model/schema/trace';
+
+import { TraceDetailData, TraceDetailDataSourceModel } from './trace-detail-data-source.model';
 
 @Model({
   type: 'api-trace-detail-data-source'
 })
-export class ApiTraceDetailDataSourceModel extends GraphQlDataSourceModel<TraceDetailData> {
-  @ModelProperty({
-    key: 'trace',
-    required: true,
-    type: PLAIN_OBJECT_PROPERTY.type
-  })
-  public trace!: Trace;
-
-  @ModelProperty({
-    key: 'attributes',
-    type: ARRAY_PROPERTY.type
-  })
-  public attributes: string[] = [];
-
-  private readonly attributeSpecBuilder: SpecificationBuilder = new SpecificationBuilder();
-
-  public getData(): Observable<TraceDetailData> {
-    return this.query<TraceGraphQlQueryHandlerService>({
-      requestType: TRACE_GQL_REQUEST,
-      traceType: this.trace[traceTypeKey],
-      traceId: this.trace[traceIdKey],
-      spanLimit: 0,
-      timeRange: this.getTimeRangeOrThrow(),
-      traceProperties: this.getTraceAttributes().map(attribute =>
-        this.attributeSpecBuilder.attributeSpecificationForKey(attribute)
-      )
-    }).pipe(mergeMap(trace => this.mapResponseObject(trace)));
-  }
-
+export class ApiTraceDetailDataSourceModel extends TraceDetailDataSourceModel {
```

Comment:
I see a trace detail data source at this path in my local. Was it changed recently?

Reply:
I renamed trace- to api-trace. That was merged yesterday. Then, this change I'm adding back a trace- and making api-trace extend that instead.

Reply:
For an api trace, what does the `traceIdKey` in `Trace` object represent? Would it be traceId or apiTraceId/entrySpanId?

Reply:
For the query to work, It should represent `apiTraceId`. Right?
If we want consistency, then probably we always make traceIdKey represent traceId and for fetching api trace detail, we use entrySpanId in the getData query below.

Reply:
Its confusing. For an `API_TRACE`, the `traceIdKey` is the same value as the `apiTraceId` which is actually the Span ID.

Reply:
oh ohkay.. I thought we were doing the mapping in the handler somewhere. but nvm

Reply:
that may be a clearer way to handle a bunch of this - we could change the trace query handler to always match traceIdKey to the real trace id and introduce an optional entrySpanId (or just leave it to the requester)